packages/delegate/src/defaultMergedResolver.ts

```diff
@@ -28,9 +42,307 @@ export function defaultMergedResolver(
+      }
+      const deferred = createDeferred();
+      missingDeferredFields.set(responseKey, deferred);
+      return deferred.promise;
+    }
+    return undefined;
+  }
+  return handleResult(parent, responseKey, context, info);
+}
+
+function handleResult(
+  parent: ExternalObject,
+  responseKey: string,
+  context: Record<string, any>,
+  info: GraphQLResolveInfo,
+) {
+  const subschema = getSubschema(parent, responseKey);
   const data = parent[responseKey];
   const unpathedErrors = getUnpathedErrors(parent);
-  const subschema = getSubschema(parent, responseKey);
 
-  return resolveExternalValue(data, unpathedErrors, subschema, context, info);
+  const resolvedData$ = resolveExternalValue(data, unpathedErrors, subschema, context, info);
+  const leftOver = getPlanLeftOverFromParent(parent);
+  // Handle possible deferred fields if any left over from the previous delegation plan is found
+  if (leftOver) {
+    if (isPromise(resolvedData$)) {
+      return resolvedData$.then(resolvedData => {
+        parent[responseKey] = resolvedData;
+        handleLeftOver(parent, context, info, leftOver);
+        return resolvedData;
+      });
+    }
+    parent[responseKey] = resolvedData$;
+    handleLeftOver(parent, context, info, leftOver);
+  }
+  return resolvedData$;
+}
+
+function handleLeftOver(
+  parent: ExternalObject,
+  context: any,
+  info: GraphQLResolveInfo,
+  leftOver: DelegationPlanLeftOver,
+) {
+  const stitchingInfo = info.schema.extensions?.['stitchingInfo'] as StitchingInfo;
+  if (stitchingInfo) {
+    for (const possibleSubschema of leftOver.nonProxiableSubschemas) {
+      const parentTypeName = info.parentType.name;
+      const selectionSet =
+        stitchingInfo.mergedTypes[parentTypeName].selectionSets.get(possibleSubschema);
+      // Wait until the parent is flattened, then check if non proxiable subschemas are satisfied now,
+      // then the deferred fields can be resolved
+      if (selectionSet) {
+        const flattenedParent$ = flattenPromise(parent);
+        if (isPromise(flattenedParent$)) {
+          flattenedParent$.then(flattenedParent => {
+            handleFlattenedParent(
+              flattenedParent,
+              possibleSubschema,
+              selectionSet,
+              leftOver,
+              stitchingInfo,
+              parentTypeName,
+              context,
+              info,
+            );
+          });
+        } else {
+          handleFlattenedParent(
+            flattenedParent$,
+            possibleSubschema,
+            selectionSet,
+            leftOver,
+            stitchingInfo,
+            parentTypeName,
+            context,
+            info,
+          );
+        }
+      }
+    }
+  }
+}
+
+function handleFlattenedParent(
+  flattenedParent: ExternalObject,
+  possibleSubschema: Subschema,
+  selectionSet: SelectionSetNode,
+  leftOver: DelegationPlanLeftOver,
+  stitchingInfo: StitchingInfo,
+  parentTypeName: string,
+  context: any,
```

Comment:
```suggestion
  context: unknown,
```

Reply:
Removed any 👍 I cannot use unknown there because I need to change a lot in other functions so it became mess in my first attempt which also might make the PR bigger 😓 
So I used `TContext`.

Reply:
Ok cool fair enough. Lets keep it then for later.